src/chainparamsbase.cpp

```diff
@@ -91,12 +91,19 @@ std::string ChainNameFromCommandLine()
 {
     bool fRegTest = gArgs.GetBoolArg("-regtest", false);
     bool fTestNet = gArgs.GetBoolArg("-testnet", false);
+    bool fMainNet = gArgs.GetBoolArg("-mainnet", false);
```

Comment:
I don't think you need to change chainparamsbase for this.

I'd say a simpler solution would be: add logic to one of the Init() functions to detect running of mainnet when not `#ifdef ENABLE_MAINNET` and then `InitError()` out.

Reply:
Makes sense.  Will revert `src/chainparamsbase.cpp` and make requested changes in `src/init.cpp` today or tomorrow.  Thanks!